### ISSUE TITLE
update to version 1.4.0 not working

### ISSUE THREAD

**Issue by andre-dietrich** [jasongilman/proto-repl#136]
Hi, after the last update I got the following error message:

Some installed packages could not be loaded because they contain native modules that were compiled for an earlier version of Atom.￼Rebuild Packages
￼Package Settings
proto-repl 1.4.0

build@ – Cannot find module '/home/andre/.atom/packages/proto-repl/node_modules/highlights/node_modules/first-mate/node_modules/oniguruma/build'

**Comment by carocad** [comment]
I can confirm this bug on my machine as well :/
The message is exactly the same and it is reported directly by atom on the right-lower corner (red bug symbol)

Proto-repl 1.4.0
Atom 1.8.0
Ubuntu 16.04

EDIT: the problem dissapeared as I recompiled proto-repl (through the recompile button displayed near the error message) and reloaded atom.

**Comment by jasongilman** [comment]
It looks like a problem with the syntax highlighting dependency. I think this could be a known atom issue but I'll take a look. I believe there's a way to fix this with an apm command.

**Comment by jasongilman** [comment]
@andre-dietrich, @carocad Can you try running `apm rebuild` in ~/.atom/packages/proto-repl to see if that fixes the problem? Either that or there should be a link to rebuild packages in the bottom right.

**Comment by carocad** [comment]
@jasongilman as I mentioned on the EDIT part of my [previous comment](https://github.com/jasongilman/proto-repl/issues/136#issuecomment-238994289), the problem was solved in my case by rebuilding proto-repl using the buttton that was near the bug-report that I obtained.

**Comment by carocad** [comment]
this bug seems to happens everytime that I update proto-repl. It started happened on v1.4.0 and has since happened for the other two updates 1.4.1 and 1.4.2. Although on my normal atom environment it can be solved by recompiling the package, once I tried to run proto-repl with `atom --dev` it crashes on activation. Here is the stacktrace:

```
Failed to activate the proto-repl package
Cannot find module 'highlights'
Hide Stack Trace
Error: Cannot find module 'highlights'
    at Module._resolveFilename (module.js:339:15)
    at Function.Module._resolveFilename (/usr/share/atom/resources/app.asar/src/module-cache.js:383:52)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/home/camilo/.atom/packages/proto-repl/lib/views/ink-console.coffee:2:14)
    at Object.<anonymous> (/home/camilo/.atom/packages/proto-repl/lib/views/ink-console.coffee:1:1)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/home/camilo/Proyectos/proto-repl/lib/repl.coffee:5:14)
    at Object.<anonymous> (/home/camilo/Proyectos/proto-repl/lib/repl.coffee:1:1)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/home/camilo/.atom/packages/proto-repl/lib/proto-repl.coffee:10:8)
    at Object.<anonymous> (/home/camilo/.atom/packages/proto-repl/lib/proto-repl.coffee:4:1)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (/usr/share/atom/resources/app.asar/src/package.js:715:27)
```

I already tried apm rebuild in both my proto-repl dev project and in the installed one but that doesn't seem to solve the problem :(

PS: I also updated atom to the latest stable version. 
Atom 1.9.8
Ubuntu 16.04

**Comment by jasongilman** [comment]
Highlights is a new dependency that I added. I have a second computer that I installed Proto REPL on and upgraded it without getting a problem. 

There might be a real problem here but we should make sure that it's not just because you do development on Proto REPL too. 

You said you have two instances of Proto REPL installed. Where are those directories located in Atom?

In the Proto REPL Development instance do you update it by doing a git pull? If so did you run npm install there? I think that might be part of the problem. 

Did you update Atom before you installed the new versions?

**Comment by carocad** [comment]
Oh I didn't know that highlights was a new dependency. I think that's where the problem lies.

As I mentioned I have the installed proto-repl and the development one. The installed one was simply fixed by atom itself but since the development one was old, the highlights module was not there, thus throwing the exception when importing it.

Syncing my local fork with your remote/master didn't help either. I think this is due to .gitignore excluding node_modules, thus the dependency was still missing.

I manage to solve the problem by manually installing highlights in my global environment via `npm install highlights`. Once installed, my `atom --dev` instance asked me to recompile proto-repl and after reloading it the problem was solved.

To answer your questions
installed: ~/.atom/packages/proto-repl
development: ~/Proyectos/proto-repl

The development one has an extra symlink to atom as per their documention

I updated atom before going to version 1.4.2 (if I remember correctly).

**Comment by moxaj** [comment]
I had a similar issue, related to the oniguruma package. I solved it by installing Visual Studio 2015 build tools, followed by running "apm config set msvs_version 2015".

**Comment by jasongilman** [comment]
You shouldn't have to install highlights that way. If you run "npm install" in the development Proto REPL folder it will install anything that's missing. It might not now though since it's a global install now. Atom sets up new packages without node_modules in git. The dependencies get installed when you upgrade or install a package through atom. If you're doing development on a package and update through git it's a manual process to run npm install when dependencies change. 

I'm no expert on package development though so I could have some of this wrong. It's just what I've figured out working on Proto REPL.

**Comment by carocad** [comment]
well I guess that this 'bug' is then only for those that have a development version of proto-repl + an installed version of it. Do @moxaj and @andre-dietrich have the same conditions?

Probably the dev version is messing with the installed one somehow. I didn't know about the manual apm install step for dev packages. I am also very new to this so thanks for the info :)

**Comment by vs192ef34** [comment]
I can confirm the issue with upgrading to 1.4.2. As @moxaj wrote above it is related to oniguruma package.

Below is the error log

```
> oniguruma@5.1.2 install C:\Users\Vlad\AppData\Local\Temp\apm-install-dir-116716-772-tchy4b\node_modules\proto-repl\node_modules\highlights\node_modules\first-mate\node_modules\oniguruma
> node-gyp rebuild


C:\Users\Vlad\AppData\Local\Temp\apm-install-dir-116716-772-tchy4b\node_modules\proto-repl\node_modules\highlights\node_modules\first-mate\node_modules\oniguruma>if not defined npm_config_node_gyp (node "C:\dev\atom\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild ) 

gypnpm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\dev\\atom\\resources\\app\\apm\\bin\\node.exe" "C:\\dev\\atom\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Vlad\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Vlad\\.atom\\.apmrc" "install" "C:\\Users\\Vlad\\AppData\\Local\\Temp\\d-116716-772-o15ddh\\package.tgz" "--target=0.37.8" "--arch=ia32"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! oniguruma@5.1.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the oniguruma@5.1.2 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the oniguruma package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls oniguruma
npm ERR! There is likely additional logging output above.
```

**Comment by jasongilman** [comment]
@vs192ef34 Did it show a Rebuild packages button? Have you tried clicking that to resolve the problem?

**Comment by meng-hui** [comment]
A new installation fails with the error like what @vs192ef34 posted. I'm using Windows 10 and Atom 1.9.9

**Comment by vs192ef34** [comment]
@jasongilman no, it didn't. 
![update-fail](https://cloud.githubusercontent.com/assets/1922310/17812629/924b7af6-6630-11e6-9920-040465a87ff3.png)

Please see the attached image, just the same error and no Rebuild button

**Comment by jasongilman** [comment]
This should be fixed now with Proto REPL 1.4.4. I changed how Proto REPL was using the highlights dependency so it removes the need for the oniguruma transitive dependency. Please let me know if there are still issues.

**Comment by vs192ef34** [comment]
Yes, now it works! Thanks @jasongilman !
